src/api/getWilayaByCode.js

```diff
@@ -0,0 +1,19 @@
+const projectWilaya = require('../utils/projections/wilayaProjection');
+
+const getWilayaByCode = data =>
+/**
+ * Takes a wilaya code (matricule) and returns its respective wilaya
+ *
+ * @example <p> Get Oran wilaya
+ * //returns ["Oran"]
+ * getWilayaByCode(31)
+ *
+ * @param { Number} mattricule The Wilaya's "matricule"
+ * @returns { Number[] | undefined } Returns matching wilaya, or undefined
+ */
+  (mattricule, projection) =>  {
+    const wilaya = data.find(w => w.mattricule === mattricule);
+    return projectWilaya(wilaya, projection);
+  };
```

Comment:
Please check if `wilaya` is defined before passing it to `projectWilaya`, you may want to do something like this:

```suggestion
// eslint-disable-next-line consistent-return
  (mattricule, projection) =>  {
    const wilaya = data.find(w => w.mattricule === mattricule);
    if (wilaya) {
      return projectWilaya(wilaya, projection);
    }
  };
```

Also add a test case for this (if the wilaya is not found and you pass an array of strings for projection, example `getWilayaByCode(mockData)(31, ['name'])`) so we ensure that this won't slip away from us next time.


Or, do you think that we should apply this check in `src/utils/projections/wilayaProjection.js` ? If so I can update my PR (#7 ) with this fix. But you may have another suggestion.